pkg/i2gw/ingress2gateway.go

```diff
@@ -136,53 +140,53 @@ func GetSupportedProviders() []string {
 	return supportedProviders
 }
 
-// MergeGatewayResources accept multiple GatewayResources and create a unique Resource struct
-// built as follows:
+// MergeGatewayIRs accepts multiple IRs and creatse a unique IR struct built
+// as follows:
 //   - GatewayClasses, *Routes, and ReferenceGrants are grouped into the same maps
 //   - Gateways may have the same NamespaceName even if they come from different
 //     ingresses, as they have a their GatewayClass' name as name. For this reason,
 //     if there are mutiple gateways named the same, their listeners are merged into
 //     a unique Gateway.
 //
 // This behavior is likely to change after https://github.com/kubernetes-sigs/gateway-api/pull/1863 takes place.
-func MergeGatewayResources(gatewayResources ...GatewayResources) (GatewayResources, field.ErrorList) {
-	mergedGatewayResources := GatewayResources{
-		Gateways:        make(map[types.NamespacedName]gatewayv1.Gateway),
+func MergeIRs(irs ...IR) (IR, field.ErrorList) {
+	mergedIRs := IR{
```

Comment:
I cant see any referecnce for merging Service IRs

Reply:
Updated.